core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -943,7 +952,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalString(routing);
             out.writeOptionalString(parent);
             out.writeLong(version);
-            
+
             out.writeByte(versionType.getValue());
             out.writeLong(autoGeneratedIdTimestamp);
             out.writeVLong(seqNo);
```

Comment:
We do run with assertions enabled; isn't it because we guard in the serialization:

```
            if (format >= FORMAT_SEQ_NO) {
                seqNo = in.readVLong();
                primaryTerm = in.readVLong();
            }
```

Either way, I think it's simpler to just use `(read|write)Long` everywhere.